doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1639 @@
+
+<a id="121-Basic-Approach"></a>
+### 1.2.1 Basic Approach
+The diagram below shows a typical EVPN Multihoming network with Multihomed (MH)  and Single-homed (SH) devices connected to EVPN VxLAN VTEPs.
+
+![](images/evpn_mh_diagram.PNG)
+
+__Figure 1: Typical EVPN Multihoming network__
+
+The devices H1, H4, and H5 are SH devices and connected to Vtep-1, Vtep-4, and Vtep-5 respectively. The devices H2 & H3 are multihomed and connected to EVPN VxLAN network with active-active redundancy using LAG. Device H2 is multihomed to Vtep-1 and Vtep-4 whereas device H3 is multihomed to Vtep-1, Vtep-2, Vtep-3, and Vtep-4.
+
+In order to configure EVPN Multihoming for the LAGs connecting multihomed devices, Type-0, Type-1, or Type-3 ESI is required to be configured on the LAG interfaces:
+- *T*ype-0: Operator configured ESI*
+- *Type-1: ESI automatically derived from LACP partner MAC*
+- *Type-3: ESI automatically derived from configured LAG system-mac address*
+
+For a given Multihomed Ethernet-Segment, the value of ES-ID should be same on all of the participating VTEPs.
+
+For simplicity, assume Type-3 ESI is configured on the PortChannel interfaces shown in the above diagram. The Vtep-1 and Vtep-4 LAG configuration will look like this:
+
+```
+sudo config interface PortChannel1 sys-mac 00:00:00:00:11:11
+sudo config interface PortChannel1 evpn-esi auto-system-mac
+
+sudo config interface PortChannel2 system-mac 00:00:00:00:22:22
+sudo config interface PortChannel2 evpn-esi auto-system-mac
+```
+
+Whereas, on Vtep-2 and Vtep-3 only PortChannel2 config is required since only PortChannel2 is multihomed to these VTEPs.
+```
+sudo config interface PortChannel2 system-mac 00:00:00:00:22:22
+sudo config interface PortChannel2 evpn-esi auto-system-mac
+```
+The Type-3 ESI value will be derived by combining the configured system-mac address on the PortChannel interface, and the PortChannel number. The system-mac is required to be same on all of the VTEPs multihoming a given LAG.
+
+The above diagram and PortChannel configurations will be used in subsequent sub-sections to explain the traffic forwarding scenarios in EVPN Multihoming network.
+
+#### 1.2.1.1 BUM forwarding in EVPN MH
+The BUM (Broadcast, Unknown Unicast, and Multicast) traffic in EVPN Multihoming network requires different handling depending on whether BUM traffic is originated from VTEP participating or not participating in the EVPN Multihoming.
+
+##### 1.2.1.1.1 Local-bias and split-horizon filtering
+If the BUM traffic originates from a device attached to an EVPN Ethernet-Segment, local-bias procedure is applied and BUM traffic is flooded on all locally attached multi-homed and single-homed devices. The BUM traffic is also replicated to the remote VTEPs, and split-horizon filtering is applied such that traffic is not replicated to any of the devices that are attached to the shared EVPN Ethernet-Segment.
```

Comment:
Can you please mention that, this is applicable to only all-active MH?

Reply:
Local bias comes with VxLAN due to fact there is no indication of incoming BUM traffic along with a easy way to identify corresponding Ethernet-Segment. In MPLS, it is possible to apply ESI-filtering over BUM traffic. Cannot be done for VxLAN. That says, the method works for other load-balancing modes including single-active and port-active. The initial design do not support them yet.